plutus-core/untyped-plutus-core/test/Evaluation/Builtins/Definition.hs

```diff
@@ -167,13 +168,22 @@ test_IdRank2 :: TestTree
 test_IdRank2 =
     testCase "IdRank2" $ do
         let res = mkConstant @Integer @DefaultUni () 0
-            -- sum (idRank2 {list} nil {integer})
+            -- > sum (idRank2 {list} nil {integer})
             term
                 = apply () (mapFun Left Scott.sum)
                 . tyInst () (apply () (tyInst () (builtin () $ Right IdRank2) Scott.listTy) Scott.nil)
                 $ integer
         typecheckEvaluateCekNoEmit def defaultBuiltinCostModelExt term @?= Right (EvaluationSuccess res)
 
+-- | Test that a builtin can be applied to a non-constant term.
+test_ScottToMetaUnit :: TestTree
+test_ScottToMetaUnit =
+    testCase "ScottToMetaUnit" $ do
+        let res = mkConstant @() @DefaultUni () ()
+            -- > scottToMetaUnit Scott.unitval
+            term = apply () (builtin () ScottToMetaUnit) Scott.unitval
```

Comment:
What happens if you pass a term that _doesn't_ have the right shape? Nothing, presumably?

Reply:
Yup, we don't check it's of the right shape. It's ensured by the type checker, though. But that's irrelevant for untyped evaluation, of course.

Reply:
In the branch where it's allowed to apply arguments of builtin functions we could've implemented

```
scottToMetaUnit f = f {()} ()
```

(provided we add handling of type application, which sounds doable), but I don't think we care about any of that to complicate the builtins machinery even more.

Reply:
Maybe worth having a test that passing something else also succeeds?

Reply:
Just to make the demonstration obvious.

Reply:
Will do, thanks.